lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -511,8 +515,11 @@ public function getRootEntities()
         foreach ($this->_dqlParts['from'] as &$fromClause) {
             if (is_string($fromClause)) {
                 $spacePos = strrpos($fromClause, ' ');
-                $from     = substr($fromClause, 0, $spacePos);
-                $alias    = substr($fromClause, $spacePos + 1);
+
+                $from = substr($fromClause, 0, $spacePos);
+                assert(class_exists($from));
```

Comment:
Likewise, there is a feature that allows to use an interface here: https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/cookbook/resolve-target-entity-listener.html I think this assertion might be wrong because of that.

Reply:
I didn't know how to solve both the static analysis error and the cs style error.

First, I added. `/** @var class-string $from */` and got a cs error.
I tried again with `@psalm-var`, seems like it's ok like this.

It might be better to not doing any assertions so far to avoid any errors with the deprecated class-alias feature.